packages/devtools_app/lib/src/service/service_manager.dart

```diff
@@ -539,6 +539,23 @@ class ServiceConnectionManager {
     await whenValueNonNull(isolateManager.mainIsolate);
     return libraryUriAvailableNow(uri);
   }
+
+  Future<String?> rootLibraryForSelectedIsolate() async {
+    if (!connectedState.value.connected) return null;
+
+    final selectedIsolateRef = isolateManager.mainIsolate.value?.id;
+    if (selectedIsolateRef == null) return null;
+
+    final selectedIsolate = await service!.getIsolate(selectedIsolateRef);
```

Comment:
If the isolate is already set to be the main isolate, it should either be in the process of becoming runnable or already runnable, meaning `IsolateManager` should have the full `Isolate` object. Consider doing something along these lines instead:

```dart
final isolateState = isolateManager.isolateState(selectedIsolateRef);
await isolateState.waitForIsolateLoad();
final rootLib = isolateState.rootInfo!.library;
```